pkg/components/rook/component_test.go

```diff
@@ -0,0 +1,221 @@
+				"rook-ceph/templates/service-monitor.yaml",
+			}
+
+			for _, ec := range expectedConfig {
+				if _, ok := m[ec]; !ok {
+					t.Fatalf("expected %q to be generated but it is not available", ec)
+				}
+			}
+		}},
+		{"OperatorTolerations", func(t *testing.T, m map[string]string) {
+			expected := []corev1.Toleration{
+				{Key: "toleration_key1", Operator: "Equal", Value: "toleration_value1", Effect: "NoSchedule"},
+				{Key: "toleration_key2", Operator: "Equal", Value: "toleration_value2", Effect: "NoSchedule"},
+			}
+			got := deploy.Spec.Template.Spec.Tolerations
+
+			if !reflect.DeepEqual(expected, got) {
+				t.Fatalf("expected: %v, got: %v", expected, got)
+			}
+		}},
+		{"EnvVars", func(t *testing.T, m map[string]string) {
+			// nolint:lll
+			subTCs := map[string]string{
+				"AGENT_TOLERATION_KEY":        "agent_toleration_key",
+				"AGENT_TOLERATION":            "NoSchedule",
+				"AGENT_TOLERATIONS":           "- effect: NoSchedule\n  key: toleration_key1\n  operator: Equal\n  value: toleration_value1\n- effect: NoSchedule\n  key: toleration_key2\n  operator: Equal\n  value: toleration_value2",
+				"DISCOVER_TOLERATION_KEY":     "discover_toleration_key",
+				"DISCOVER_TOLERATION":         "NoSchedule",
+				"DISCOVER_TOLERATIONS":        "- effect: NoSchedule\n  key: toleration_key1\n  operator: Equal\n  value: toleration_value1\n- effect: NoSchedule\n  key: toleration_key2\n  operator: Equal\n  value: toleration_value2",
+				"CSI_PROVISIONER_TOLERATIONS": "- effect: NoSchedule\n  key: toleration_key1\n  operator: Equal\n  value: toleration_value1\n- effect: NoSchedule\n  key: toleration_key2\n  operator: Equal\n  value: toleration_value2",
+				"CSI_PLUGIN_TOLERATIONS":      "- effect: NoSchedule\n  key: other_node_key1\n  operator: Equal\n  value: other_node_value1",
+
+				// TODO: Fix the ordering in implementation of convertNodeSelector. Once that is not
+				// consistent, we can start testing following cases.
+				// https://github.com/kinvolk/lokomotive/issues/942.
+				// "AGENT_NODE_AFFINITY":           "storage_node_key1=storage_node_value1; storage_node_key2=storage_node_value2;",
+				// "DISCOVER_AGENT_NODE_AFFINITY":  "storage_node_key1=storage_node_value1; storage_node_key2=storage_node_value2;",
+				// "CSI_PROVISIONER_NODE_AFFINITY": "storage_node_key1=storage_node_value1; storage_node_key2=storage_node_value2;",
+				// "CSI_PLUGIN_NODE_AFFINITY":      "node_key1=node_value1; node_key2=node_value2;",
+			}
+
+			envs := deploy.Spec.Template.Spec.Containers[0].Env
+			envVars := make(map[string]string, len(envs))
+			for _, en := range envs {
+				envVars[en.Name] = en.Value
+			}
+
+			for k, v := range subTCs {
+				k, v := k, v
+				t.Run(k, func(t *testing.T) {
+					gotVal, ok := envVars[k]
+					if !ok {
+						t.Fatalf("expected: %q env", k)
+					}
+
+					if gotVal != v {
+						t.Fatalf("expected value of env %q:%q, got %q:%q", k, v, k, gotVal)
```

Comment:
```suggestion
						t.Fatalf("Expected env var %q:%q, got %q:%q", k, v, k, gotVal)
```

We are printing the key *and* the value, not just the value. The text should match.